Library/Homebrew/os/linux/elf.rb

```diff
@@ -78,6 +79,16 @@ def dynamic_elf?
     end
   end
 
+  def corrupted_elf?
+    # elf files that throw the following message when parsed with readelf:
+    # Warning: possibly corrupt ELF header
+    # it has a non-zero program header offset, but no program headers
+    # Example: the early-section-headers.so unittest file from the rust package
+    Utils.popen_read("readelf", "-d", to_path, err: :out)
+         .strip
```

Comment:
The `.strip` here is probably not needed, unless there's white space to the left of the string.